src/tools/rust-analyzer/crates/base-db/src/lib.rs

```diff
@@ -1,4 +1,9 @@
 //! base_db defines basic database traits. The concrete DB is defined by ide.
+
+#![cfg_attr(feature = "in-rust-tree", feature(rustc_private))]
+#[cfg(all(feature = "in-rust-tree", test))]
+extern crate rustc_driver as _;
```

Comment:
Will we need to add this prelude for any new crate we add, or only for those that have (unit) tests?

Reply:
Only for new crates that have unit tests AFAIK. What I did basically was to tag all the crates that had unit tests (which complained about not finding `rustc_lexer` and friends).